ttnn/cpp/ttnn/operations/eltwise/binary/binary.cpp

```diff
@@ -113,10 +113,14 @@ auto preprocess_inputs(
 
         // repeats second if it is smaller
         if (first_shape.rank() == 4 and second_shape.rank() == 4 and first_shape[0] > second_shape[0]) {
-            tt::log_warning(tt::LogOp, "Using repeat op to broadcast batch dim");
             Shape repeats(std::array<uint32_t, 4>{first_shape[0], 1, 1, 1});
             second = ttnn::repeat(second, repeats);
         }
+        // repeats second if it is smaller
+        if (first_shape.rank() == 4 and second_shape.rank() == 4 and first_shape[1] > second_shape[1]) {
+            Shape repeats(std::array<uint32_t, 4>{1, first_shape[1], 1, 1});
```

Comment:
Isn't it a bug? I think we should pass `first_shape[1]/second_shape[1]` (also check whether `first_shape[1]` divisible by `second_shape[1]`)

Same applies to repeat for batch dimension.

Reply:
Seems like pytorch doesn't broadcast if another dimension is not 1. Maybe the proper solution is to throw exception if dimension we trying to broadcast is not equal to 1. 
<img width="847" alt="Screenshot 2024-10-15 at 10 25 50 PM" src="https://github.com/user-attachments/assets/ef721f0f-da3f-4f91-986f-d7000b628931">